### PR TITLE
Add access by index

### PR DESCRIPTION
**Pull request by panther99**
This adds an option to do DELETE request by using indexing in the path.

For example:

```
/api/laptops/0/byindex/shipment/2/byindex
```

`byindex` after the parameter will tell the resolver that we're looking for an item with the given index in an array instead of an item with the same ID as the given parameter.

### CODE REVIEW

File: tests/jsondb.test.js [hunk 1]
```diff
@@ -126,6 +126,14 @@ Deno.test("methodDelete", () => {
     data: { ...json, laptops: json.laptops.slice(1) },
   });
 
+  jsonCopy = JSON.parse(JSON.stringify(json));
```
Comment (mk0y) [inline tests/jsondb.test.js:129]: Just `{ ...json }` instead?

Comment (panther99) [reply tests/jsondb.test.js:129]: It can do only one-level copy though, here we need a deep copy.

File: jsondb.js [hunk 2]
```diff
@@ -199,6 +199,39 @@ const applyGetMethod = U.when(
...
+    if (U.isObject(current)) {
+      current = current[pathItem];
+      finalPath.push(pathItem);
+    } else if (U.isArray(current)) {
+      const index = Number(pathItem);
+      const maybeItem = current[Number(pathItem)];
+      if (isNaN(index) || maybeItem == null) {
+        return [];
+      }
+      current = current;
```
Comment (mk0y) [inline jsondb.js]: Why this?

File: jsondb.js [hunk 3]
```diff
@@ -199,6 +199,39 @@ const applyGetMethod = U.when(
...
+  for (
+    let pathItem = null, pathCopy = [...lensPath], current = { ...json };
+    (pathItem = pathCopy.shift());
+  ) {
+    if (U.isObject(current)) {
+      current = current[pathItem];
+      finalPath.push(pathItem);
+    } else if (U.isArray(current)) {
+      const index = Number(pathItem);
+      const maybeItem = current[Number(pathItem)];
```
Comment (mk0y) [inline jsondb.js]: Here can go `index`?

File: jsondb.js (hunk 2 above)
Comment (panther99) [reply jsondb.js]: A mistake, it should be `current = maybeItem`.

File: jsondb.js (hunk 3 above)
Comment (panther99) [reply jsondb.js]: Yes, left accidentally.

**Pull request merged**